api/adaptive_load/config.proto

```diff
@@ -0,0 +1,57 @@
+// Protos for selecting and configuring plugins for the adaptive load
+// controller infrastructure.
+
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "google/protobuf/any.proto";
+
+// Selects and configures a StepController plugin, which computes the next load
+// amount (e.g. RPS) and checks for convergence to the optimal load, based on
+// feedback metrics.
+//
+// For available plugin names, see step_controller_impl.proto.
+message StepControllerConfig {
+  // Name used to locate the plugin in the Envoy registry. Required.
+  string name = 1;
+  // Plugin-specific config. Required.
+  google.protobuf.Any typed_config = 2;
```

Comment:
I recommend using `TypedExtensionConfig` for all your extension configs.

Reply:
Will do. "I did not realize that was an option!"

Reply:
Done